src/roadmap/common.py

```diff
@@ -87,54 +89,183 @@ async def query_rbac(
     return data.get("data", [{}])
 
 
-async def check_inventory_access(
+def _get_group_list_from_resource_definition(resource_definition: dict) -> list[str]:
+    if attributeFilter := resource_definition.get("attributeFilter"):
+        if attributeFilter.get("key") != "group.id":
+            raise HTTPException(501, detail="Invalid value for attributeFilter.key in RBAC response.")
+        op = attributeFilter.get("operation")
+        if op not in ("in", "equal"):
+            raise HTTPException(501, detail="Invalid value for attributeFilter.operation in RBAC response.")
+        val = attributeFilter.get("value")
+        if op == "in":
+            if not isinstance(val, list):
+                raise HTTPException(501, detail="Did not receive a list for attributeFilter.value in RBAC response.")
+        else:
+            if not isinstance(val, str):
+                raise HTTPException(501, detail="Did not receive a str for attributeFilter.value in RBAC response.")
+            val = [val]
+
+        group_list = val
+        # Validate that all values in the filter are UUIDs.
+        try:
+            for gid in group_list:
+                # It is possible that the group id is None, which means
+                # permission to access a host who belongs to a group which
+                # has its "ungrouped" attribute set to True.
+                # TODO: Check if this None check can be removed once the Kessel Phase 0 migration is complete.
+                # https://issues.redhat.com/browse/RHINENG-16842
+                if gid is not None:
+                    UUID(gid)
+        except (ValueError, TypeError):
+            logger.warning(f"RBAC attributeFilter contained erroneous UUID: '{gid}'")
+            raise HTTPException(501, detail="Received invalid UUIDs for attributeFilter.value in RBAC response.")
```

Comment:
If possible, narrow the scope of this try except.
```suggestion
        for gid in group_list:
            # It is possible that the group id is None, which means
            # permission to access a host who belongs to a group which
            # has its "ungrouped" attribute set to True.
            # TODO: Check if this None check can be removed once the Kessel Phase 0 migration is complete.
            # https://issues.redhat.com/browse/RHINENG-16842
            if gid is not None:
                try:
                    UUID(gid)
                except (ValueError, TypeError):
                    logger.warning(f"RBAC attributeFilter contained erroneous UUID: '{gid}'")
                    raise HTTPException(501, detail="Received invalid UUIDs for attributeFilter.value in RBAC response.")
```

Reply:
fixed